lib/galaxy/webapps/galaxy/api/tools.py

```diff
@@ -488,7 +489,7 @@ def conversion(self, trans: GalaxyWebTransaction, tool_id, payload, **kwd):
         else:
             if input_src == "hdca":
                 target_history = self.hdca_manager.get_dataset_collection_instance(
-                    trans, instance_type="history", id=input_id
+                    trans, instance_type=DatasetCollectionInstanceType.history, id=input_id
```

Comment:
Should we instead change `DatasetCollectionInstanceType` in `lib/galaxy/schema/schema.py` to be `Literal["history", "library"]` ?